contracts/SoulBoundToken/BasicSBT.sol

```diff
@@ -0,0 +1,215 @@
+        require(hasSoul(_soul), "Soul does not exists");
+        require(
+            msg.sender == _soul,
+            "Only users have rights to delete their data"
+        );
+        delete souls[_soul];
+        delete _owners[_totalSBT];
+        emit Burn(_soul);
+    }
+
+    /**
+     * @dev Updates the mapping of address to attribute.
+     * Only the owner address is able to update the information.
+     *
+     * However, projects can have it such that users can propose changes for the contract owner to update.
+     */
+    function updateSBT(address _soul, Soul memory _soulData)
+        public
+        onlyOwner
+        returns (bool)
+    {
+        require(hasSoul(_soul), "Soul does not exist");
+        souls[_soul] = _soulData;
+        emit Update(_soul);
+        return true;
+    }
+
+    /**
+     * @dev Returns the soul data of `identity, url` for the given address
+     */
+    function getSBTData(address _soul)
+        public
+        view
+        virtual
+        validAddress(_soul)
+        returns (string memory, string memory)
+    {
+        return (souls[_soul].identity, souls[_soul].url);
+    }
+
+    /**
+     * @dev Validates if the _soul is associated with the valid data in the corresponding address.
+     * By checking if the `_soulData` given in parameter is the same as the data in the struct of mapping.
+     */
+    function validateAttribute(address _soul, Soul memory _soulData)
+        public
+        view
+        virtual
+        returns (bool)
+    {
+        require(hasSoul(_soul), "Soul does not exist");
+        (string memory _identity, string memory _url) = getSBTData(_soul);
+
+        return
+            compareString(_soulData.identity, _identity) &&
+            compareString(_soulData.url, _url);
+    }
+
+    /**
+     * @dev Gets the total count of SBT.
+     */
+    function totalSBT() public view virtual returns (uint256) {
+        return _totalSBT;
+    }
+
+    /**
+     * @dev Returns if two strings are equal
+     */
+    function compareString(string memory a, string memory b)
+        internal
+        pure
+        virtual
+        returns (bool)
+    {
+        return compareMemory(bytes(a), bytes(b));
+    }
+
+    /**
+     * @dev Returns if two memory arrays are equal
+     */
+    function compareMemory(bytes memory a, bytes memory b)
+        internal
+        pure
+        virtual
+        returns (bool)
+    {
+        return (a.length == b.length) && (keccak256(a) == keccak256(b));
+    }
+
+    /**
+     * @dev Returns whether SBT exists for a given address.
+     *
+     * SBT start existing when they are minted (`mint`),
+     * and stop existing when they are burned (`burn`).
+     */
+    function hasSoul(address _soul)
+        public
+        view
+        virtual
+        validAddress(_soul)
+        returns (bool)
+    {
+        // TODO: check with JH if we should go w this check or using IERC721 check
```

Comment:
If we use this check - after it is burnt - it will still be empty string?

Reply:
yup, however i tried it and does not work. 

If the address does not exists, it will default to an empty string

Reply:
removed that specific implementation due to line , as it is not working

Reply:
So what's the conclusion for this? Now we are not making any checks? 🤔

Reply:
using this check instead

```
(string memory _identity, string memory _url) = getSBTData(_soul);
        return bytes(_identity).length > 0 && bytes(_url).length > 0;
``